docs/developer/products/troubleshooting.mdx

```diff
@@ -0,0 +1,31 @@
+---
+title: Products troubleshooting guide
+sidebar_label: Troubleshooting
+description: Learn how to troubleshoot product-related issues in Saleor.
+---
+
+### Product is not visible in the storefront
+
+Follow this checklist if a product is not visible in the Storefront or API:
+
+1. **Product is published**: Enable `Product is published` and `Available for purchase` for the respective channel. Depending on the storefront implementation, it might be necessary to disable `Hide in product listings`.
+2. **Variant is enabled**: At least one `Variant` exists and is set to `Available` for the respective channel. `Price` needs to be higher than zero.
+3. **Use correct channel slug**: When querying products, use the correct channel slug instead of channel names, which can contain uppercase letters and white space.
+
+### Product can't be added to an order
+
+Availability of products depends on the customer's location; follow these steps to troubleshoot:
+
+1. Orders are channel-specific; ensure the product is available in the respective channel by following [these steps](#product-is-not-visible-in-the-storefront).
+2. Add customer address to the order. Depending on the destination address and the channel, the product might not be available.
+3. Find the `Shipping zone` that should be used for the customer's `Shipping` and `Billing` address.
+
+- Shipping zone should have assigned `Country` matching customer address.
+- The shipping zone should have assigned a `Channel` that matches the order's `Channel`.
+- Shipping zone has at least one `Warehouse` enabled.
+
+4. In the product variant settings, check if the `Warehouses` of the shipping zone have stock. Only warehouses enabled in the shipping zone are considered for stock availability.
```

Comment:
It's worth mentioning that even products without stock tracking require existence of the stock.

Reply:
added